### DISCUSSION TITLE
Multigraph Data Structure coding for GDAS NAS

### DISCUSSION THREAD

**Discussion by buttercutter** [networkx/networkx#4838] [Q&A]
how to code the following graph structure ?
which kind of data structure is suitable in this case for both forward inference and backward propagation ?
Would https://networkx.org/documentation/stable/tutorial.html#multigraphs be suitable ?

![image](https://user-images.githubusercontent.com/3324659/119596474-cdb75e00-be11-11eb-9da6-2e169b44706a.png)

https://github.com/D-X-Y/AutoDL-Projects/issues/99#issuecomment-835713058

https://gist.github.com/promach/b6f526c56e20f029d68e6f9041c3f5c0#file-gdas-py-L106-L107

**Comment by MridulS** [comment]
checkout [networkx.MultiDiGraph](https://networkx.org/documentation/stable/reference/classes/multidigraph.html#networkx.MultiDiGraph) too if you want `directionality` in the network.

**Reply by buttercutter** [reply]
However, does the multigraph have support for both forward inference and backward propagation operations ?

**Reply by MridulS** [reply]
No, you would need to implement them or maybe find another library which has done it already :)
